### PR TITLE
Closes #19517

### LINKED ISSUE #19517: Add date to unstable version string
Currently the Geth version strings look like `Geth/v1.8.27-stable-4bcc0a37/linux-amd64/go1.10.4` and `Geth/v1.9.0-unstable-def1b0d7/linux-amd64/go1.12.1`.

Whilst the stable releases are mostly meaningful and we can easily put a date on them (or if not a date, at least we can compare one stable release to another chronologically), we cannot do that for unstable builds without looking up the commits. This makes it hard for example to see which nodes are hard-fork ready on the Rinkeby stats page and which are not.

We should add a date suffix `-YYYYMMDD` after the commit hash for unsable builds.

### PR DESCRIPTION
**Pull request by Hackdom**
Add date to unstable release

### CODE REVIEW

File: params/version.go [hunk 1]
```diff
@@ -33,12 +33,14 @@ var Version = func() string {
 }()
 
 // VersionWithMeta holds the textual version string including the metadata.
-var VersionWithMeta = func() string {
+var VersionWithMeta, Unstable = func() (string, bool) {
```
Comment (karalabe) [inline params/version.go]: I don't think you need to separate out this extra variable `Unstable`. You can just compare `params.Version` with `"unstable"` at the call site.

File: params/version.go [hunk 2]
```diff
@@ -33,12 +33,14 @@ var Version = func() string {
...
 	v := Version
+	unstable := false
 	if VersionMeta != "" {
 		v += "-" + VersionMeta
+		unstable = true
```
Comment (karalabe) [inline params/version.go]: This is not true, `VersionMeta` will either be `"stable"` or `"unstable"`, but generally not empty.

File: cmd/utils/flags.go [hunk 3]
```diff
@@ -102,6 +102,9 @@ func NewApp(gitCommit, usage string) *cli.App {
 	if len(gitCommit) >= 8 {
 		app.Version += "-" + gitCommit[:8]
 	}
+	if params.Unstable {
```
Comment (karalabe) [inline cmd/utils/flags.go]: Please use `params.VersionMeta != "stable"`, simpler, cleaner.

**Comment by Hackdom** [comment]
got it... fixes enroute

**Comment by Hackdom** [comment]
simplified version is up. thanks for looking :)

**Comment by Hackdom** [comment]
got it

**Comment by Hackdom** [comment]
Question ref adding it into the VersionWithCommit helper. The code would look like this:
```
func VersionWithCommit(gitCommit string, gitDate string) string {
	vsn := VersionWithMeta
	if len(gitCommit) >= 8 {
		vsn += "-" + gitCommit[:8]
	}
	if VersionMeta != "stable" {
		app.Version += "-" + gitDate
	}
	return vsn
}
```
Which would label anything not "stable". Is that what you'd like, or should we keep this scoped specifically to "unstable"?

**Comment by Hackdom** [comment]
Let me know if that is what you're thinking. Also, amended the string split in the second commit. That can be removed if you don't like it.

**Comment by Hackdom** [comment]
just saw this wasn't working still. another update pushed.

**Comment by Hackdom** [comment]
Still not showing... I'm trying to stay out of the build flags, standby.

**Comment by Hackdom** [comment]
ok, done. way more painful than necessary.

**Comment by Hackdom** [comment]
Opened in https://github.com/ethereum/go-ethereum/pull/19522

**Pull request closed without merging**